client/my-sites/current-site/index.jsx

```diff
@@ -90,14 +91,12 @@ class CurrentSite extends Component {
 	};
 
 	render() {
-		const { selectedSite, translate, anySiteSelected, rtlOn } = this.props;
+		const { selectedSite, translate, anySiteSelected, rtlOn, isRequestingAllSites } = this.props;
 
-		if ( ! anySiteSelected.length ) {
+		if ( ! anySiteSelected.length || ( ! selectedSite && isRequestingAllSites ) ) {
```

Comment:
Again, this will show a placeholder even if the all sites list is available.

The problem is that after this patch, we can have two situations we are unable to distinguish from each other:
- `state.sites.items` contains the full list of all sites
- `state.sites.items` contains just one or two sites we individually requested, but the full list is not known

We can solve this in two ways:
1. Introduce a boolean flag `state.sites.hasAllSitesList`
2. store an array with the site list IDs in Redux: `state.sites.allSitesList`. The `state.sites.items` would be an unsorted pile of site items, and `allSitesList` would be an array with sorted list of all sites. The `state.posts` already uses a similar technique. There is a map of post IDs to post objects, and a lot of ID lists that correspond to various post queries.

Then various selectors would check if the data they need are available. For example:
- `getSite` would just check `state.sites.items`
- `hasJetpackSites` needs the whole list, so it returns `null` if `hasAllSitesList` is `false` or if `allSitesList` is `null`

Reply:
Do you think the second approach has any advantages for `sites`? It looks to me like a lot of refactoring for not much of benefit.

I'm in favor of introducing `hasAllSitesList` flag.

Reply:
I agree, `hasAllSitesList` looks perfectly fine.